src/js/plugins/tts/plugin.tts.js

```diff
@@ -0,0 +1,253 @@
+        super_.call(this, options);
+
+        if (this.options.enableTtsPlugin) {
+            this.ttsEngine = new FestivalTTSEngine({
+                server: options.server,
+                bookPath: options.bookPath,
+                onLoadingStart: this.showProgressPopup.bind(this, 'Loading audio...'),
+                onLoadingComplete: this.removeProgressPopup.bind(this),
+                onDone: this.ttsStop.bind(this),
+                beforeChunkPlay: this.ttsBeforeChunkPlay.bind(this)
+            });
+            this.ttsHilites = [];
+        }
+    };
+})(BookReader.prototype.setup);
+
+BookReader.prototype.init = (function(super_) {
+    return function() {
+        if (this.options.enableTtsPlugin) {
+            // Bind to events
+            this.bind(BookReader.eventNames.PostInit, function(e, br) {
+                br.$('.BRicon.read').click(function(e) {
+                    br.ttsToggle();
+                    return false;
+                });
+                br.ttsEngine.init();
+            });
+
+            // This is fired when the hash changes by one of the other plugins!
+            // i.e. it will fire every time the page changes -_-
+            // this.bind(BookReader.eventNames.stop, function(e, br) {
+            //     this.ttsStop();
+            // }.bind(this));
+        }
+        super_.call(this);
+    };
+})(BookReader.prototype.init);
+
+
+// Extend buildMobileDrawerElement
+BookReader.prototype.buildMobileDrawerElement = (function (super_) {
+    return function () {
+        var $el = super_.call(this);
+        if (this.options.enableTtsPlugin && this.ttsEngine.isSupported) {
+            $el.find('.BRmobileMenu__moreInfoRow').after($(
+                "    <li>"
+                +"      <span>"
+                +"        <span class=\"DrawerIconWrapper \"><img class=\"DrawerIcon\" src=\""+this.imagesBaseURL+"icon_speaker_open.svg\" alt=\"info-speaker\"/></span>"
+                +"        Read Aloud"
+                +"      </span>"
+                +"      <div>"
+                +"        <span class='larger'>Press to toggle read aloud</span> <br/>"
+                +"        <button class='BRicon read'></button>"
+                +"      </div>"
+                +"    </li>"
+            ));
+        };
+        return $el;
+    };
+})(BookReader.prototype.buildMobileDrawerElement);
+
+// Extend initNavbar
+BookReader.prototype.initNavbar = (function (super_) {
+    return function () {
+        var $el = super_.call(this);
+        var readIcon = '';
+        if (this.options.enableTtsPlugin && this.ttsEngine.isSupported) {
+            $("<button class='BRicon read js-tooltip'></button>").insertAfter($el.find('.BRpage .BRicon.thumb'));
+        }
+        return $el;
+    };
+})(BookReader.prototype.initNavbar);
+
+// ttsToggle()
+//______________________________________________________________________________
+BookReader.prototype.ttsToggle = function () {
```

Comment:
Since you've got several methods in here that are tts-specific, what do you think of namespacing them instead of putting them all at the top-level? ie `BookReader.tts.toggle()`, `BookReader.tts.start()`, etc.

Reply:
https://github.com/internetarchive/bookreader/issues/138 :) In the interest of keeping the scope of this small (lol), I wanted to avoid doing that as part of this PR. It is planned, but this first refactor took longer than anticipated, so I may have to punt that for later :/